packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -2483,6 +2483,18 @@ moduleFor('Components test: curly components', class extends RenderingTest {
     this.assertText('Jackson');
   }
 
+  ['@test injecting an unknown service raises an exception']() {
+    this.registerComponent('foo-bar', {
+      ComponentClass: Component.extend({
+        missingService: inject.service()
+      })
+    });
+
+    throws(() => {
```

Comment:
That `validateInjections` method is only called from here: 
https://github.com/emberjs/ember.js/blob/9b3d581e9b706ae636691b1fff199a86c793b53d/packages/container/lib/container.js#L505-L513

Which is inside of a `runInDebug`...